plugins/ceph_osd_plugin.py

```diff
@@ -87,23 +90,40 @@ def get_stats(self):
                 osd_data['in'] += 1
             else:
                 osd_data['out'] += 1
-    
+
         return data
 
+    def get_stats_via_tool(self):
+        """Retrieves stats using a subprocess."""
+        raw_output = subprocess.check_output(
+            ['ceph', 'osd', 'dump', '--format', 'json', '--cluster ', self.cluster])
+        return json.loads(raw_output)
+
+    @staticmethod
+    def get_stats_via_rest():
+        """Retrieves stats using a request."""
+        response = requests.get("http://127.0.0.1:7766/api/v0.1/osd/dump",
```

Comment:
macro

Reply:
I guess the IP and port must be given as part of the configuration

Reply:
I'll add them